src/Core/Binding/IBindingConfigProvider.cs

```diff
@@ -44,9 +44,11 @@ public interface IBindingConfigProvider
     /// For C++ it will be in a json file in a Sonar-specific format</remarks>
     public interface IBindingConfigFile
     {
+        string FilePath { get; }
```

Comment:
This is main change: we will now store the filePath of the ruleset inside the BindingConfigFile.